Rearranging & adding flags (#735)
graphprotocol/docs: website/pages/en/publishing/publishing-a-subgraph.mdx

- [Curators](/network/curating) to begin curating it.
- [Indexers](/network/indexing) to begin indexing it.

<VideoEmbed youtube="satdwdbvzAQ" />

Check out the list of [supported networks](/developing/supported-networks).

## Publishing from Subgraph Studio

1. Go to the [Subgraph Studio](https://thegraph.com/studio/) dashboard
2. Click on the **Publish** button
3. Your subgraph will now be visible in [Graph Explorer](https://thegraph.com/explorer/).

All published versions of an existing subgraph can:

- Be published to Arbitrum One. [Learn more about The Graph Network on Arbitrum](/arbitrum/arbitrum-faq).

- Index data on any of the [supported networks](/developing/supported-networks), regardless of the network on which the subgraph was published.

### Updating metadata for a published subgraph

- After publishing your subgraph to the decentralized network, you can update the metadata anytime in Subgraph Studio.
- Once you’ve saved your changes and published the updates, they will appear in Graph Explorer.
- It's important to note that this process will not create a new version since your deployment has not changed.

## Publishing from the CLI

> As of version 0.73.0, you can also publish your subgraph with the `graph-cli`.

Reviewer: ```suggestion
As of version 0.73.0, you can also publish your subgraph with the `graph-cli`.
```
```suggestion
As of version 0.73.0, you can also publish your subgraph with the [`graph-cli`](https://github.com/graphprotocol/graph-tooling/tree/main/packages/cli).
```

1. Open the `graph-cli`.

> 2. Use the following commands: `graph publish` then `graph build && graph codegen`

Reviewer: first you want to codegen -> build and finally publish the built artifacts.

```suggestion
2. Use the following commands: `graph codegen && graph build` then `graph publish`.
```

3. A window will open, allowing you to connect your wallet, add metadata, and deploy your finalized subgraph to a network of your choice.

![cli-ui](/img/cli-ui.png)

### Customizing your deployment

You can upload your subgraph build to a specific IPFS node and further customize your deployment with the following flags:

```

> USAGE
>   $ graph publish [SUBGRAPH-MANIFEST] [-h] [-i <value>] [--ipfs-hash <value>] [--webapp-url <value>] [--subgraph-id <value>] [--protocol-name <value>]
> 
> FLAGS
>   -h, --help            Show CLI help.
>   -i, --ipfs=<value>    [default: https://api.thegraph.com/ipfs/api/v0] Upload build results to an IPFS node.
>   --ipfs-hash=<value>   IPFS hash of the subgraph manifest to deploy.
>   --webapp-url=<value>  [default: https://cli.thegraph.com/publish] URL of the web UI you want to use to deploy.
>   --subgraph-id=<value>     ID of the subgraph to update.
>   --protocol-name=<value>   Name of the protocol for the subgraph update.
> 

Reviewer: ```suggestion
USAGE
  $ graph publish [SUBGRAPH-MANIFEST] [-h] [--protocol-network arbitrum-one|arbitrum-sepolia --subgraph-id <value>] [-i <value>] [--ipfs-hash <value>] [--webapp-url
    <value>]

FLAGS
  -h, --help                   Show CLI help.
  -i, --ipfs=<value>           [default: https://api.thegraph.com/ipfs/api/v0] Upload build results to an IPFS node.
  --ipfs-hash=<value>          IPFS hash of the subgraph manifest to deploy.
  --protocol-network=<option>  [default: arbitrum-one] The network to use for the subgraph deployment.
                               <options: arbitrum-one|arbitrum-sepolia>
  --subgraph-id=<value>        Subgraph ID to publish to.
  --webapp-url=<value>         [default: https://cli.thegraph.com/publish] URL of the web UI you want to use to deploy.

```

> ```
> USAGE
>   $ graph publish [SUBGRAPH-MANIFEST] [-h] [-i <value>] [--ipfs-hash <value>] [--webapp-url <value>] [--subgraph-id <value>] [--protocol-name <value>]
> 
> FLAGS
>   -h, --help            Show CLI help.
>   -i, --ipfs=<value>    [default: https://api.thegraph.com/ipfs/api/v0] Upload build results to an IPFS node.
>   --ipfs-hash=<value>   IPFS hash of the subgraph manifest to deploy.
>   --webapp-url=<value>  [default: https://cli.thegraph.com/publish] URL of the web UI you want to use to deploy.
>   --subgraph-id=<value>     ID of the subgraph to update.
>   --protocol-name=<value>   Name of the protocol for the subgraph update.
> 
> ```

Reviewer: ```suggestion
```
USAGE
  $ graph publish [SUBGRAPH-MANIFEST] [-h] [--protocol-network arbitrum-one|arbitrum-sepolia --subgraph-id <value>] [-i <value>] [--ipfs-hash <value>] [--webapp-url
    <value>]

FLAGS
  -h, --help                   Show CLI help.
  -i, --ipfs=<value>           [default: https://api.thegraph.com/ipfs/api/v0] Upload build results to an IPFS node.
  --ipfs-hash=<value>          IPFS hash of the subgraph manifest to deploy.
  --protocol-network=<option>  [default: arbitrum-one] The network to use for the subgraph deployment.
                               <options: arbitrum-one|arbitrum-sepolia>
  --subgraph-id=<value>        Subgraph ID to publish to.
  --webapp-url=<value> 

```
```

![Signal from Explorer](/img/signal-from-explorer.png)

> Learn more about [Curating](https://thegraph.com/docs/en/network/curating/) on The Graph Network.

Reviewer 2: ```suggestion
Learn more about [Curating](/network/curating/) on The Graph Network.
```
